execution.bs

```diff
@@ -3173,67 +3196,91 @@ Insert this section as a new subclause between Header `<stop_token>` synopsis <b
     requires {
-      { T::stop_possible() } -> boolean-testable;
-    } &&
-    (!T::stop_possible());
+      { bool_constant&lt;T::stop_possible()>{} } -> same_as&lt;false_type>;
+    };
 </pre>
 
-2. Let `t` and `u` be distinct object of type `T`. The type `T` models `stoppable_token` only if:
+<div class="ed-note">LWG directed me to replace `T::stop_possible()` with `t.stop_possible()` because
+of the recent `constexpr` changes in [[P2280r2|P2280R2]]. However, even with those changes, a nested
+requirement like `requires (!t.stop_possible())`, where `t` is an argument in the requirement-parameter-list, is ill-formed according to
+<a href="http://eel.is/c++draft/expr.prim.req#nested-2.sentence-1">[expr.prim.req.nested/p2]</a>:
+
+> A local parameter shall only appear as an unevaluated operand within the constraint-expression.
+
+This is the subject of core issue [[cwg2517|2517]].
+</div>
+
+2. Let `t` and `u` be distinct, valid objects of type `T`. The type `T` models `stoppable_token` only if:
 
-    1. All copies of a `stoppable_token` reference the same logical shared stop state and shall report values consistent with each other.
+    1. If `t.stop_possible()` evaluates to `false` then, if `u` references the same logical shared stop state, `u.stop_possible()` shall also subsequently evaluate to `false` and `u.stop_requested()` shall also subsequently evaluate to `false`.
 
-    2. If `t.stop_possible()` evaluates to `false` then, if `u`, references the same logical shared stop state, `u.stop_possible()` shall also subsequently evaluate to `false` and `u.stop_requested()` shall also subsequently evaluate to `false`.
+    2. If `t.stop_requested()` evaluates to `true` then, if `u` references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
 
-    3. If `t.stop_requested()` evaluates to `true` then, if `u`, references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
+3. Let `t` and `u` be distinct, valid objects of type `T` and let `init` be an
+    object of type `Initializer`. Then for some type `CB`, the type `T` models
+    `stoppable_token_for<CB, Initializer>` only if:
 
-    4. Given a callback-type, CB, and a callback-initializer argument, `init`, of type `Initializer` then constructing an instance, `cb`, of type `T::callback_type<CB>`, passing `t` as the first argument and `init` as the second argument to the constructor, shall,
-        if `t.stop_possible()` is `true`, construct an instance, `callback`, of type `CB`, direct-initialized with `init`, and register callback with `t`’s shared stop state such that callback will be invoked with an empty argument list if a stop request is made on
-        the shared stop state.
+    1. Constructing an instance, `cb`, of type `T::callback_type<CB>`, passing
```

Comment:
Direct non-list initializating `cb` of type `T::callback_type<CB>` from `t` and `init`.